### PR TITLE
Add ability to register model

### LINKED ISSUE #147: [Deployment] Add ability to register a model
**Describe the solution you'd like**
For a given run, we would like to be able to register these model as something we would like to be able to deploy.
This issue is part of a larger deployment epic. We most likely will need a backing table with the following columns:

- Id
- RunId

With the following methods
- `RegisterModel(Guid runId)`
- 'RegisterModel(Guid runId, string filePath)'

**Additional context**
image

### PR DESCRIPTION
**Pull request by aslotte**
## Resolves
Fixes #147

### CODE REVIEW

**Comment by aslotte** [comment]
This is ready for a review! Thanks all for helping out!

File: src/MLOps.NET/MLOpsBuilder.cs [hunk 1]
```diff
@@ -34,7 +35,7 @@ public IMLOpsContext Build()
         public MLOpsBuilder UseMetaDataRepositories(IDbContextFactory contextFactory)
         {
             this.experimentRepository = new ExperimentRepository(contextFactory);
-            this.runRepository = new RunRepository(contextFactory);
+            this.runRepository = new RunRepository(contextFactory, new Clock());
```
Comment (mbrdev) [inline src/MLOps.NET/MLOpsBuilder.cs:38]: Should this change be reverted since we didn't use it in the end?

Comment (aslotte) [reply src/MLOps.NET/MLOpsBuilder.cs:38]: I think we can keep it. It will be useful once we start adding unit tests to the `RunRepository`

**Comment by aslotte** [comment]
@mbrdev / @Brett-Parker - just realized Cosmos DB also most likely needed the common entity configuration that we added, so moved it to the common context. Ran our CosmosDB integration tests locally again and it all passed.

**Comment by aslotte** [comment]
Thanks all!

**Pull request merged**